packages/quill/src/app/screens/Accounts.tsx

```diff
@@ -76,35 +74,16 @@ const Account = ({ account }: { account: IAccount }) => {
 };
 
 export const Accounts = () => {
+  const dispatch = useDispatch();
   const accounts = useSelector(getAccounts);
 
   useNavigation(ROUTE_PATHS.SETTINGS);
 
-  if (accounts.length === 0) {
-    return (
-      <>
-        <Container pt="4">
-          <Box sx={{ textAlign: 'center' }}>
-            <Image alt="Wallet" src={wallet} />
-          </Box>
-          <Box maxWidth="80%" mx="auto" sx={{ textAlign: 'center' }}>
-            <SubHeading mt="3" mb="2">
-              {translateRaw('ACCOUNTS_EMPTY_HEADER')}
-            </SubHeading>
-            <Body variant="muted">{translateRaw('ACCOUNTS_EMPTY_BODY')}</Body>
-          </Box>
-        </Container>
-        <PanelBottom variant="clear">
-          <LinkApp href={ROUTE_PATHS.GENERATE_ACCOUNT}>
-            <Button mb="3">{translateRaw('GENERATE_ACCOUNT')}</Button>
-          </LinkApp>
-          <LinkApp href={ROUTE_PATHS.ADD_ACCOUNT}>
-            <Button variant="inverted">{translateRaw('ADD_ACCOUNT')}</Button>
-          </LinkApp>
-        </PanelBottom>
-      </>
-    );
-  }
+  useEffect(() => {
+    if (accounts.length === 0) {
+      dispatch(push(ROUTE_PATHS.SETUP_ACCOUNT));
+    }
+  });
```

Comment:
Should this be dependant on `accounts.length`?

Reply:
Yep. Good call